### ISSUE TITLE
[Bug] [Zeta] Config variable substitution 不一致

### ISSUE THREAD

**Issue by xuqi1633** [apache/seatunnel#6840]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

来自官网案例: https://seatunnel.apache.org/docs/2.3.5/concept/config/#config-variable-substitution

变量替换 和传入的变量不一致

### SeaTunnel Version

2.3.5

### SeaTunnel Config

```conf
env {
  job.mode = "BATCH"
  job.name = ${jobName}
  parallelism = 2
}

source {
  FakeSource {
    result_table_name = ${resName}
    row.num = ${rowNum}
    string.template = ${strTemplate}
    int.template = [20, 21]
    schema = {
      fields {
        name = ${nameType}
        age = "int"
      }
    }
  }
}

transform {
    sql {
      source_table_name = "fake"
      result_table_name = "sql"
      query = "select * from "${resName}" where name = '"${nameVal}"' "
    }

}

sink {
  Console {
     source_table_name = "sql"
     username = ${username}
     password = ${password}
     blankSpace = ${blankSpace}
  }
}
```

### Running Command

```shell
./bin/seatunnel.sh -c <this_config_file> 
-i jobName='st var job' 
-i resName=fake 
-i rowNum=10 
-i strTemplate=['abc','d~f','h i'] 
-i nameType=string 
-i nameVal=abc 
-i username=seatunnel=2.3.1 
-i password='$a^b%c.d~e0*9(' 
-i blankSpace='2023-12-26 11:30:00' 
-e local
```

### Error Exception

```log
no error
```

### Screenshots

![企业微信截图_1fef571b-888a-4190-b5af-247959c5bbe5](https://github.com/apache/seatunnel/assets/70441327/15d9e938-bb8f-40ff-844c-45fc54284b70)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Hisoka-X** [comment]
@liunaijie would you mind take a look?

**Comment by liunaijie** [comment]
if the variables has blank space, will has issue. will check on this.

**Comment by xuqi1633** [comment]
@liunaijie can you review above pr

**Comment by Hisoka-X** [comment]
closed by #7099
